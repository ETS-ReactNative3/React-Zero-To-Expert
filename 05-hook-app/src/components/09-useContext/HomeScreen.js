import React, { useContext } from 'react'
import { UserContext } from './UseContext'

const HomeScreen = () => {
  // const userContext = useContext(UserContext)
  // console.log(userContext)
  const { user } = useContext(UserContext)
  console.log(user)
  return (
    <>
      <h1>HomeScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
    </>
  )
}

export default HomeScreen
