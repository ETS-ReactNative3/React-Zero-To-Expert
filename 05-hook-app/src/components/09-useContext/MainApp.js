import React from 'react'
import AppRouter from './AppRouter'
import { UserContext } from './UseContext'

const MainApp = () => {
  const user = {
    id: 1234,
    name: 'elfGod',
    email: 'elfgod@elfgod.com',
  }

  return (
    <UserContext.Provider value={user}>
      <AppRouter />
    </UserContext.Provider>
  )
}

export default MainApp
