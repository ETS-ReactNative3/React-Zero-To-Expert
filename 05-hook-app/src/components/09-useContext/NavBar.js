import React from 'react'
import { Link, NavLink } from 'react-router-dom'

const NavBar = () => {
  return (
    <nav className='navbar navbar-expand-sm navbar-dark bg-dark'>
      <div className='container-fluid'>
        <Link to='/' className='navbar-brand'>
          useContext
        </Link>
        <div className='collapse navbar-collapse' id='navbarNavAltMarkup'>
          <div className='navbar-nav'>
            <NavLink
              className={(isActive) =>
                'nav-link' + (!isActive ? ' unselected' : '')
              }
              to='/'
            >
              Home
            </NavLink>
            <NavLink
              className={(isActive) =>
                'nav-link' + (!isActive ? ' unselected' : '')
              }
              to='/about'
            >
              About
            </NavLink>
            <NavLink
              className={(isActive) =>
                'nav-link' + (!isActive ? ' unselected' : '')
              }
              to='/login'
            >
              Login
            </NavLink>
          </div>
        </div>
      </div>
    </nav>
  )
}

export default NavBar

{
  /* <navbar>
  <ul>
    <li>
      <Link to='/'>Home</Link>
    </li>
    <li>
      <Link to='/about'>About</Link>
    </li>
    <li>
      <Link to='/login'>Login</Link>
    </li>
  </ul>
</navbar> */
}
