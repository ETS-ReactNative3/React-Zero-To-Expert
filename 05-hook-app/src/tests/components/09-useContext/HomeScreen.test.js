import React from 'react'
import { mount } from 'enzyme'
import HomeScreen from '../../../components/09-useContext/HomeScreen'
import { UserContext } from '../../../components/09-useContext/UseContext'
describe('Pruebas  en <HomeScreen />', () => {
  const user = {
    name: 'ElfGod',
    email: 'elfgod@gmail.com',
  }

  const wrapper = mount(
    <UserContext.Provider
      value={{
        // user: user
        user,
      }}
    >
      <HomeScreen />
    </UserContext.Provider>
  )

  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
