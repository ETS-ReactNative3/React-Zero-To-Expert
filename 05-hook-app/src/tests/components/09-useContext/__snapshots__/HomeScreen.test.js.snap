// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas  en <HomeScreen /> debe de mostrarse correctamente 1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <hr />,
  <pre>
    {
   "name": "ElfGod",
   "email": "elfgod@gmail.com"
}
  </pre>,
]
`;
